### PR TITLE
Spelling fix & update docs with date_time_render_option behaviour

### PR DESCRIPTION
**Pull request by alifeee**
This started as a spelling change - I updated "formating" to "formatting for `gspread.utils.DateTimeOption`

I added a test for `date_time_render_option` as there did not seem to be one. I found that the conflict behaviour of this kwarg with `value_render_option` was different to how it was in the docs:

Docs:

image

Actual behaviour:

If both are set, and `value_render_option` is set to `ValueRenderOption.formatted`, then an API error is thrown by Google's API.

Thus, I have included this in the test, and updated the documentation to reflect this.

In summary:

- Fixed spelling `DateTimeOption.formated_string` -> `DateTimeOption.formated_string`
- Added test `test_get_all_values_date_time_render_options`
  - Added test cassette
- Updated documentation for `date_time_render_option`
  - changed `note` to `warning`
  - explained that if `value_render_option` is set to `ValueRenderOption.formatted`, a `gspread.exceptions.APIError` will be raised

### CODE REVIEW

**Review by lavigne958 — COMMENTED** [review]
Hi thank you very much for this contribution, it is complete, with tests and doc.

I am sorry I cannot merge that fix in the master branch yet. the change will break peoples code as it changes the user input. We can merge it in the feature branch I have that for the future 6.0.0 release: feature/release_6_0_0

File: gspread/utils.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 ValueInputOption = namedtuple("ValueInputOption", ["raw", "user_entered"])(
     "RAW", "USER_ENTERED"
 )
-DateTimeOption = namedtuple("DateTimeOption", ["serial_number", "formated_string"])(
+DateTimeOption = namedtuple("DateTimeOption", ["serial_number", "formatted_string"])(
```
[the remark this answers is not in the document]
Comment (alifeee) [reply gspread/utils.py:37]: That's very understandable :)

I have changed the pull request to pull into feature/release_6_0_0

[the remark this answers is not in the document]
Comment (lavigne958) [reply gspread/utils.py:37]: thank you for the changes

**Review by lavigne958 — APPROVED** [review]
This is very neat thank you for this contribution.
